lib/resource.js

```diff
@@ -57,19 +57,71 @@ export default class PodiumClientResource {
     }
 
     async fetch(incoming = {}, reqOptions = {}) {
-        if (!utils.validateIncoming(incoming)) throw new TypeError('you must pass an instance of "HttpIncoming" as the first argument to the .fetch() method');
+        if (!utils.validateIncoming(incoming))
+            throw new TypeError(
+                'you must pass an instance of "HttpIncoming" as the first argument to the .fetch() method',
+            );
         const outgoing = new HttpOutgoing(this.#options, reqOptions, incoming);
 
+        if (this.#options.excludeBy) {
+            /**
+             * @type {string[] | undefined}
+             */
+            const exclucedDeviceTypes = this.#options.excludeBy.deviceType;
+            if (Array.isArray(exclucedDeviceTypes)) {
+                const deviceTypeHeader =
+                    incoming.request.headers['x-podium-device-type'];
+
+                for (let i = 0; i < exclucedDeviceTypes.length; i += 1) {
+                    const shouldSkip =
+                        exclucedDeviceTypes[i] === deviceTypeHeader;
+                    if (shouldSkip) {
+                        return new Response({
+                            headers: {},
+                            content: '',
+                            css: [],
+                            js: [],
+                            redirect: '',
+                        });
+                    }
+                }
+            }
+        }
+
+        if (this.#options.includeBy) {
+            /**
+             * @type {string[] | undefined}
+             */
+            const includeDeviceTypes = this.#options.includeBy.deviceType;
+            if (Array.isArray(includeDeviceTypes)) {
+                const deviceTypeHeader =
+                    incoming.request.headers['x-podium-device-type'];
+
+                const shouldRequest =
+                    !deviceTypeHeader ||
+                    includeDeviceTypes.includes(deviceTypeHeader);
+
+                if (!shouldRequest) {
+                    return new Response({
+                        headers: {},
+                        content: '',
+                        css: [],
+                        js: [],
+                        redirect: '',
+                    });
+                }
+            }
+        }
+
```

Comment:
should includeBy take precedence over excludeBy? Should there be any validation of cases? Eg. If you defined the same device type in both excludeBy and includeBy. Should one be able to use both excludeBy and includeBy or should it be if/else? ie. Either you use a black list or a white list but not both.

Reply:
Good questions, what do you think?

I'll admit we added `includeBy` mostly for completeness. `excludeBy` is the main case here, so if it opens a whole can of worms maybe we just drop `includeBy` 😅